### PR TITLE
[NodeJs] Get module info by reading package.json and add

### PR DESCRIPTION
**Pull request by balajikris**
package name and version to user agent header for all auto rest
generated node packages.

This is part of work for https://github.com/Azure/azure-sdk-for-node/issues/2076. 
This change relies on runtime updates by PR https://github.com/Azure/azure-sdk-for-node/pull/2084 to fetch info from package.json.

### CODE REVIEW

**Comment by balajikris** [comment]
I think my source branch is out of date. I'll pull from master and resubmit.

**Comment by balajikris** [comment]
I've rebased my branch to latest master and still build fails. I haven't touched this file, can someone help me understand this failure and how to correct it?

~~~
/home/travis/build/Azure/autorest/src/core/AutoRest.Core/Settings.cs(93,24): warning CS0169: The field 'Settings._input' is never used [/home/travis/build/Azure/autorest/src/core/AutoRest.Core/AutoRest.Core.csproj]
~~~

File: src/generator/AutoRest.NodeJS.Azure/CodeGeneratorJsa.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace AutoRest.NodeJS.Azure
 {
     public class CodeGeneratorJsa : NodeJS.CodeGeneratorJs
     {
-        private const string ClientRuntimePackage = "ms-rest-azure version 1.15.0";
+        private const string ClientRuntimePackage = "ms-rest-azure version 1.15.6";
```
Comment (balajikris) [inline src/generator/AutoRest.NodeJS.Azure/CodeGeneratorJsa.cs]: don't pull this in yet, I'm still working on publishing these updated packages.

**Comment by amarzavery** [comment]
@azuresdkci  Test this please!

**Comment by olydis** [comment]
@balajikris nice, now CI is failing due to test files not having been generated (whooops, the travis version of that check passed...). run `gulp regenerate` and this PR should be fine! :)

**Comment by balajikris** [comment]
@azuresdkci Test this please!

**Comment by balajikris** [comment]
Tests are failing because they're referencing an older version of msrest which doesnt have the below function (which was added in 1.15.5 and above).
**Logs**:
~~~
/home/travis/build/Azure/autorest/src/generator/AutoRest.NodeJS.Tests/Expected/AcceptanceTests/CustomBaseUri/autoRestParameterizedHostTestClient.js:49
  var packageInfo = this.getPackageJsonInfo(__dirname);
                         ^
TypeError: this.getPackageJsonInfo is not a function
    at new AutoRestParameterizedHostTestClient 
~~~

**Comment by balajikris** [comment]
@azuresdkci Test this please!

**Pull request merged**